src/main/java/fr/insee/pogues/configuration/rest/AuthenticationUserHelper.java

```diff
@@ -0,0 +1,28 @@
+package fr.insee.pogues.configuration.rest;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class AuthenticationUserHelper implements AuthenticationHelper {
+    @Override
+    public String getUserToken() {
+        if(getAuthenticationPrincipal() instanceof JwtAuthenticationToken auth) {
+            return auth.getToken().getTokenValue();
+        } else {
+            log.warn("Cannot retrieve token for the user.");
+            return null;
+        }
```

Comment:
It was my first try, but when whe use healthcheck end-point with authentication, these methode want to retrieve token to send healthcheck request -> error, because end-point is public (not auth required).

Reply:
I want to retrieve token conditionnaly to public-urls